src/ConstructiveSolidGeometry/plotting/SurfacePrimitives/ConeMantle.jl

```diff
@@ -1,17 +1,52 @@
-@recipe function f(cm::ConeMantle, n = 40; subn = 10)
-    ls = lines(cm)
-    linecolor --> :black
-    @series begin
-        label --> "Cone Mantle"
-        ls[1]
+function mesh(cm::ConeMantle{T}; n = 30)::Mesh{T} where {T}
+    rbot, rtop = get_r_limits(cm)
+    φMin, φMax = get_φ_limits(cm)
+    
+    f = (φMax - φMin)/(2π)
+    n = Int(ceil(n*f))
+    φ = range(φMin, φMax, length = n+1)
+    z = range(-cm.hZ, cm.hZ, length = 2)
+
+    X::Array{T,2} = [radius_at_z(cm,z[j])*cos(φ_i) for φ_i in φ, j in 1:length(z)]
+    Y::Array{T,2} = [radius_at_z(cm,z[j])*sin(φ_i) for φ_i in φ, j in 1:length(z)]
+    Z::Array{T,2} = [j for i in 1:length(φ), j in z]
+    if cm.rotation != one(SMatrix{3, 3, T, 9})
+        cm.rotation*Mesh{T}(X,Y,Z) + cm.origin
+    else
+        Mesh{T}(X,Y,Z) + cm.origin
     end
-    for i in 2:length(ls)
+end
+
+@recipe function f(cm::ConeMantle, n = 40; subn = 10)
+    colorbar := false
+    if haskey(plotattributes, :seriestype) && plotattributes[:seriestype] == :surface
+            m = mesh(cm, n = n)
+            if occursin("GRBackend", string(typeof(plotattributes[:plot_object].backend)))
+                @series begin
+                    label --> "Cone Mantle"
+                    polymesh(m)
+                end
+            else
+                @series begin
+                    label --> "Cone Mantle"
+                    m
+                end
+            end
+    else
+        ls = lines(cm)
+        linecolor --> :black
         @series begin
-            label := nothing
-            ls[i]
+            label --> "Cone Mantle"
+            ls[1]
+        end
+        for i in 2:length(ls)
+            @series begin
+                label := nothing
+                ls[i]
+            end
         end
     end
-     if !haskey(plotattributes, :show_normal) || plotattributes[:show_normal]
+    if !haskey(plotattributes, :show_normal) || plotattributes[:show_normal]
```

Comment:
is `!haskey` intended, i.e. default to plot it if not specified?
I would personally prefer `haskey` and only plot it if explicitly specified.

Reply:
This line is not my code, but I agree

Reply:
Yes, we should set the default to `false`.